### PR TITLE
Fix create_vectorscan for the new virtual awg 

### PR DESCRIPTION
**Pull request by eendebakpt**
Allow the `create_vectorscan` method to work with the new virtual awg object.

### CODE REVIEW

File: src/qtt/measurements/scans.py [hunk 1]
```diff
@@ -2226,20 +2226,24 @@ def create_vectorscan(virtual_parameter, g_range=1, sweeporstepdata=None, remove
     if sweeporstepdata is None:
```
[the remark this answers is not in the document]
Comment (QFer) [reply src/qtt/measurements/scans.py]: Not part of your changes but:
1. sweep_or_step_data would be better readable. Also change docstring when this is changed.

File: src/qtt/measurements/scans.py [hunk 2]
```diff
@@ -2226,20 +2226,24 @@ def create_vectorscan(virtual_parameter, g_range=1, sweeporstepdata=None, remove
...
-                   for p, r in virtual_parameter.comb_map if round(r, 5) != 0])
+        active_parameters = dict([(p.name, r)
+                   for p, r in virtual_parameter.comb_map if round(r, ndigits=5) != 0])
         if remove_slow_gates:
             try:
-                for gate in list(pp.keys()):
-                    if not station.awg.awg_gate(gate):
-                        pp.pop(gate, None)
-
+                if 'awg' in station.components:
```
Comment (QFer) [inline src/qtt/measurements/scans.py]: station parameter is not in docstring

File: src/qtt/measurements/scans.py [hunk 3]
```diff
@@ -2226,20 +2226,24 @@ def create_vectorscan(virtual_parameter, g_range=1, sweeporstepdata=None, remove
...
                 warnings.warn(f'error when removing slow gate {gate} from scan data')
     else:
-        pp = {virtual_parameter.name: 1}
+        active_parameters = {virtual_parameter.name: 1}
     sweeporstepdata = {'start': start, 'range': g_range,
-                       'end': start + g_range, 'param': pp}
+                       'end': start + g_range, 'param': active_parameters}
     if step is not None:
         sweeporstepdata['step'] = step
     return sweeporstepdata
```
Comment (QFer) [inline src/qtt/measurements/scans.py]: Would be really nice if you could add a unit test for this method in test_scans.py with remove slow gates true...

File: src/qtt/measurements/scans.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (peendebak) [reply src/qtt/measurements/scans.py]: Updated the code. Why the parameter is there, but not used, I do not know

File: src/qtt/measurements/scans.py (hunk 3 above)
Comment (peendebak) [reply src/qtt/measurements/scans.py]: Out of scope for the PR

**Comment by peendebak** [comment]
Addressed the comments. I  reformatted the code, so the number of changes is high now. @QFer

File: src/qtt/measurements/scans.py [hunk 4]
```diff
@@ -958,13 +964,13 @@ def _convert_scanjob_vec(self, station, sweeplength=None, steplength=None, stepv
                     sweepdata['step'] = sweepdata['range'] / sweeplength
                 else:
                     sweepdata['step'] = (
-                        sweepdata['end'] - sweepdata['start']) / sweeplength
+                                                sweepdata['end'] - sweepdata['start']) / sweeplength
```
Comment (QFer) [inline src/qtt/measurements/scans.py]: sweepdata['step'] = (sweepdata['end'] - sweepdata['start']) / sweeplength

File: src/qtt/measurements/scans.py [hunk 5]
```diff
@@ -611,13 +617,13 @@ def makeScanjob(sweepgates, values, sweepranges, resolution):
     nx = len(sweepgates)
     step = sweepranges[0] / resolution[0]
     stepdata = {'gates': [sweepgates[0]], 'start': values[0]
-                - sweepranges[0] / 2, 'end': values[0] + sweepranges[0] / 2,
+                                                   - sweepranges[0] / 2, 'end': values[0] + sweepranges[0] / 2,
```
Comment (QFer) [inline src/qtt/measurements/scans.py]: stepdata = {'gates': [sweepgates[0]],
                'start': values[0] - sweepranges[0] / 2,
                'end': values[0] + sweepranges[0] / 2,

File: src/qtt/measurements/scans.py [hunk 6]
```diff
@@ -611,13 +617,13 @@ def makeScanjob(sweepgates, values, sweepranges, resolution):
...
                 'step': step}
     sj['stepdata'] = stepdata
     if nx == 2:
         step = sweepranges[1] / resolution[1]
         sweepdata = {'gates': [sweepgates[1]], 'start': values[1]
-                     - sweepranges[1] / 2, 'end': values[1] + sweepranges[0] / 2,
+                                                        - sweepranges[1] / 2, 'end': values[1] + sweepranges[0] / 2,
```
Comment (QFer) [inline src/qtt/measurements/scans.py]: sweepdata = {'gates': [sweepgates[1]],
                     'start': values[1] - sweepranges[1] / 2,
                     'end': values[1] + sweepranges[0] / 2,

**Pull request merged**